rest/v8/guild.ts

```diff
@@ -25,15 +27,20 @@ import type {
 	Nullable,
 	StrictPartial,
 	StrictRequired,
+	UnionToIntersection,
 } from '../../utils/internals';
 import type { RESTPutAPIChannelPermissionJSONBody } from './channel';
 
 export interface APIGuildCreateOverwrite extends RESTPutAPIChannelPermissionJSONBody {
 	id: number | string;
 }
 
+export type APIGuildChannelResolvable = Exclude<APIChannel, APIDMChannel | APIGroupDMChannel>;
 export type APIGuildCreatePartialChannel = StrictPartial<
-	Pick<APIChannel, 'type' | 'topic' | 'nsfw' | 'bitrate' | 'user_limit' | 'rate_limit_per_user'>
+	Pick<
+		UnionToIntersection<APIGuildChannelResolvable>,
```

Comment:
what exactly does this achieve?

Reply:
So the issue here was that because `APIChannel` is a union, `Pick` would say that some of those keys didn't exist. The intersection conversion makes it like the old `APIChannel` so those keys always exist.

Reply:
That is... Hacky but impressive. Nice